Mathlib/MeasureTheory/Constructions/BorelSpace/Basic.lean

```diff
@@ -707,3 +707,41 @@ protected theorem comap (μ : Measure β) [IsFiniteMeasureOnCompacts μ] (f : α
 end MeasureTheory.Measure.IsFiniteMeasureOnCompacts
 
 end BorelSpace
+
+section Support
+
+open MeasureTheory Measure
+
+open scoped Topology
+
+variable {X : Type*} [TopologicalSpace X] [MeasurableSpace X]
+
+variable {μ : Measure X}
+
+lemma support_restrict_subset_closure [OpensMeasurableSpace X] {s : Set X} :
+  (μ.restrict s).support ⊆ closure s := by
+  intro x hx
+  simp only [mem_closure_iff_nhds]
+  rw [(nhds_basis_opens x).forall_iff (fun _ _ h ↦ Set.Nonempty.mono (by gcongr))]
+  intro U ⟨hxU, hU⟩
+  by_cases H : (s ∩ U).Nonempty
+  · exact Set.inter_nonempty_iff_exists_right.mpr H
+  · have h_restr : (μ.restrict s) U = μ (U ∩ s) := by
+      simp [Measure.restrict_apply hU.measurableSet, Set.inter_comm]
+    rw [(nhds_basis_opens x).mem_measureSupport] at hx
+    exact MeasureTheory.nonempty_of_measure_ne_zero
+      (ne_of_gt (h_restr ▸ hx U ⟨hxU, hU⟩))
+
+lemma support_restrict_subset_support {s : Set X} : (μ.restrict s).support ⊆ μ.support := by
+   intro x hx
+   rw [mem_support_iff_forall] at *
+   intro U hU
+   exact lt_of_lt_of_le (hx U hU) <| restrict_apply_le _ _
+
+lemma support_restrict_subset_closure_inter_support [OpensMeasurableSpace X] {s : Set X} :
+    (μ.restrict s).support ⊆ closure s ∩ μ.support := by
```

Comment:
I would prove this version directly, without the two preceding lemmas. In order to make working with this slightly easier, I suggest adding this lemma:
```lean
lemma mem_support_restrict [OpensMeasurableSpace X] {s : Set X} {x : X} :
    x ∈ (μ.restrict s).support ↔ ∃ᶠ u in (𝓝[s] x).smallSets, 0 < μ u := by
  sorry
```
In addition, I would prove the following lemma giving a "lower bound" on the size of the support of a restriction:
```lean
lemma interior_inter_support [OpensMeasurableSpace X] {s : Set X} :
    interior s ∩ μ.support ⊆ (μ.restrict s).support := by
  sorry
```